yacl/crypto/primitives/tpre/tpre.h

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Chengfang Financial Technology Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef YACL_CRYPTO_PRIMITIVES_TPRE_TPRE_H_
+#define YACL_CRYPTO_PRIMITIVES_TPRE_TPRE_H_
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "yacl/crypto/primitives/tpre/capsule.h"
+#include "yacl/crypto/primitives/tpre/keys.h"
+
+namespace yacl::crypto {
+/**
+ * The TPRE class encapsulates the methods of threshold proxy
+ * re-encryption, including encryption, decryption, re-encryption and
+ * multi-party decryption
+ *
+ * Here, let's assume that there are N+2 participants, including Alice, the data
+ * holder, Bob, the data requester, and multiple proxy re-encryption nodes
+ * P_0,..., P_{N-1}.
+ *
+ * Assume m is a data to shared. Alice and Bob will perform the following
+ * calculation steps:
+ *
+ * 1. Alice generates the ciphertext c of plaintext m via
+ *    Encrypt(pk_A, m,...)；Here，Alice can decryt c to m by Decrypt(sk_A,
+ *    c,...)
+ *
+ * 2. Alice entrusts P_1,..., P_t re-encrypts c to obtain capsule set =
+ *    {cfrag_1,...,cfrag_t}. Only t Proxys participate here, reflecting the idea
+ *    of threshold. In the re-encryption phase, you need to use the
+ *    re-encryption key The re-encryption key is generated by calling the
+ *    re-encryption key generation algorithm Keys::GenerateReKey by entering
+ *    Alice's private key and Bob's public key.
+ *
+ * 3. Inputs capsule set and secret key of Bob, decrypts the capsule set to
+ *    plaintext.
+ *
+ * For the description of capsule, you can see in the Capsule class.
+ */
+class TPRE {
+ public:
+  //   TPRE(int curve_id);
```

Comment:
Can you add a comment here? or you can simply delete the unused code.

```suggestion
  // TODO (@Cryptographer63): support more ecc curves
  // TPRE(int curve_id);
```